web/routes/eventsub.js

```diff
@@ -0,0 +1,317 @@
+module.exports = (async function() {
+    const crypto = require('crypto');
+    const tools = require('./../../tools/tools.js');
+    const sql = require('./../../sql/index.js');
+    const Redis = require('./../../tools/redis.js');
+        
+    // Notification request headers
+    const TWITCH_MESSAGE_ID = 'Twitch-Eventsub-Message-Id'.toLowerCase();
+    const TWITCH_MESSAGE_TIMESTAMP = 'Twitch-Eventsub-Message-Timestamp'.toLowerCase();
+    const TWITCH_MESSAGE_SIGNATURE = 'Twitch-Eventsub-Message-Signature'.toLowerCase();
+    const MESSAGE_TYPE = 'Twitch-Eventsub-Message-Type'.toLowerCase();
+
+    // Notification message types
+    const MESSAGE_TYPE_NOTIFICATION = 'notification';
+    const MESSAGE_TYPE_REVOCATION = 'revocation';
+
+    // Prepend this string to the HMAC that's created from the message
+    const HMAC_PREFIX = 'sha256=';
+    
+    /**
+     * @typedef { Object } EventSubSubcription
+     * @property {string} id
+     * @property {string} status
+     * @property {string} type
+     * @property {string} version
+     * @property {number} cost
+     * @property {object} condition
+     * @property {string} condition.broadcaster_user_id
+     * @property {object} transport
+     * @property {string} transport.method
+     * @property {string} transport.callback
+     * @property {string} created_at
+     */
+
+    /**
+     * @typedef { Object } EventSubNotification
+     * @property { EventSubSubcription } subscription
+     * @property { string= } challenge
+     * @property { Object } event
+     */
+
+    /**
+     * @callback EventSubCallback
+     * @param { EventSubNotification } notification
+     * @returns { Promise<import('./../../tools/redis.js').PubSubResponse | void> }
+     */
+
+    const express = require('express');
+    const router = express.Router();
+
+    router.use(express.raw({
+        type: 'application/json'
+    }));
+    
+    /* /eventsub */
+    router.post('/', async (req, res) => {
+        if (req.headers[TWITCH_MESSAGE_SIGNATURE] === undefined) {
+            res.status(400).end();
+            return;
+        }
+
+        const hmac = HMAC_PREFIX + getHmac(getSecret(), getHmacMessage(req));
+    
+        if (!verifyMessage(hmac, req.headers[TWITCH_MESSAGE_SIGNATURE])) {
```

Comment:
@hotbear1110 I need you to verify that this works using twitch-cli
Command is: `twitch event verify streamup -s <secret> -F http://127.0.0.1:3000/eventsub`
Where secret is TWITCH_SECRET in `.env`

Reply:
Windows Issue